packages/headless/src/composables/comboBox/createComboBox.ts

```diff
@@ -0,0 +1,119 @@
+import { computed, ref, type Ref } from "vue";
+import { createId } from "../../utils/id";
+import { createBuilder, computeIterated } from "../../utils/builder";
+
+export const createComboBox = createBuilder(
+  ({
+    isExpanded,
+    activeKey,
+    onToggle,
+    onSelect,
+    onFirst,
+    onLast,
+    onNext,
+    onPrevious,
+  }: {
+    isExpanded: Ref<boolean>;
+    activeKey: Ref<string | undefined>;
+    onToggle: () => void;
+    onSelect: (key: string) => void;
+    onFirst: () => void;
+    onLast: () => void;
+    onNext: () => void;
+    onPrevious: () => void;
+  }) => {
+    const inputValue = ref("");
+    const inputValid = ref(true);
+    const controlsId = createId("comboBox-control");
+    const labelId = createId("comboBox-label");
+
+    const descendantKeyIdMap: Record<string, string | undefined> = {};
+
+    const getOptionId = (key: string) =>
+      descendantKeyIdMap[key] ?? (descendantKeyIdMap[key] = createId("comboBox-option"));
+
+    const handleInput = (event: Event) => {
+      const inputElement = event.target as HTMLInputElement;
```

Comment:
Should be check if `event.target instanceof HTMLInputElement` to catch cases where the composable is used incorrectly?